src/Http/Controller/SearchController.php

```diff
@@ -7,18 +7,50 @@
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
+use App\Domain\Course\Entity\Technology;
+use App\Http\Normalizer\TechnologyPathNormalizer;
+
+
 class SearchController extends AbstractController
 {
     /**
      * @Route("/search", name="search")
      */
-    public function search(Request $request, SearchInterface $search): Response
-    {
+    public function search(
+      Request $request,
+      SearchInterface $search,
+      TechnologyPathNormalizer $technologyPathNormalizer
+    ): Response {
         $q = $request->query->get('q');
 
-        return $this->render('pages/search.html.twig', [
-            'q' => $q,
-            'results' => $search->search($q, [])['hits'],
-        ]);
+        // Si q n'est pas vide
+        if ( !empty( $q ) ) {
+
+          // on commence par chercher une techno correspondante
+          $technology = $this->getDoctrine()
+              ->getRepository(Technology::class)
+              ->findOneBy( ['name' => $q] );
```

Comment:
Je pense qu'il est préférable d'utiliser l'injection de dépendance pour le dépôt. Ce serait meilleur de créer une fonction dans le dépôt également comme `findByName(string $name)` qui récupère la technologie qui correspond avec la requête.
```suggestion
          // On recherche une technologie qui correspond avec la requête
          $technology = $technologyRepository->findByName($q);
```